56699 twentytwelve

metodiew opened the pull request:
Removing IE specific checks from Twenty Twelve default theme. Keeping the `js/html5.js` as per the note from [this comment](https://core.trac.wordpress.org/ticket/56699#comment:8)

Working with @Neychok on the ticket, so we can split the work on the default themes.

[Trac ticket: 56699](https://core.trac.wordpress.org/ticket/56699)

metodiew committed:
twentytwelve theme: removing instances of old IE fixes and patches

metodiew committed:
restore src/wp-content/themes/twentytwelve/js/html5.js

sabernhardt commented on src/wp-content/themes/twentytwelve/header.php:
```diff
@@ -9,25 +9,12 @@
-<html class="ie ie8" <?php language_attributes(); ?>>
-<![endif]-->
-<!--[if !(IE 7) & !(IE 8)]><!-->
-<html <?php language_attributes(); ?>>
```
The page still needs an `html` tag.

metodiew replied:
Good catch, deleted by mistake, restored, thanks!

sabernhardt commented on src/wp-content/themes/twentytwelve/functions.php:
```diff
@@ -201,10 +201,6 @@ function twentytwelve_scripts_styles() {
 	wp_enqueue_style( 'twentytwelve-block-style', get_template_directory_uri() . '/css/blocks.css', array( 'twentytwelve-style' ), '20230213' );
-
-	// Loads the Internet Explorer specific stylesheet.
-	wp_enqueue_style( 'twentytwelve-ie', get_template_directory_uri() . '/css/ie.css', array( 'twentytwelve-style' ), '20150214' );
```
The stylesheet handle needs to stay in the `functions.php` file in case someone refers to it, but `wp_enqueue_style` can be replaced with `wp_register_style`.

metodiew replied:
Replaced with `wp_register_style`, is there anything else to be optimized before resolving the comment?

sabernhardt replied:
I think that would work. Thanks!

sabernhardt replied:
Oops! I was wrong. Line 207 should stay, too.
`$wp_styles->add_data( 'twentytwelve-ie', 'conditional', 'lt IE 9' );`

metodiew replied:
I was thinking the same, but went blindly with the change :sweat_smile: 
It has been restored.

sabernhardt commented on src/wp-content/themes/twentytwelve/readme.txt:
```diff
@@ -36,10 +36,6 @@ GNU General Public License for more details.
 
 Twenty Twelve Theme bundles the following third-party resources:
 
-HTML5 Shiv v3.7.0, Copyright 2014 Alexander Farkas
```
The credit belongs in the readme file because the script is still part of the package.

sabernhardt commented on src/wp-content/themes/twentytwelve/css/ie.css:
Please do not delete _any_ files. Child themes can still refer to them.

sabernhardt reviewed (changes requested):
This PR removes items that need to remain: the `ie.css` file, the `twentytwelve-ie` handle, an `html` tag, and the credit for the HTML5 Shiv (which is still included in the package).

metodiew committed:
Apply the feedback notes:
- Restore the html tag deleted by mistake
- Restore the credits
- Restore the deleted file

metodiew committed:
Apply the feedback notes:
- Restore the html tag deleted by mistake
- Restore the credits
- Restore the deleted file

metodiew commented:
Thanks for the feedback @sabernhardt ! Applied the notes and changes, can you take a look is there anything else? if not, I'll open a new PR for the other themes, following the ticket updates. Thanks in advance!

metodiew committed:
Restore wp_styles->add_data

sabernhardt commented on src/wp-content/themes/twentytwelve/functions.php:
```diff
@@ -203,7 +203,7 @@ function twentytwelve_scripts_styles() {
 	wp_enqueue_style( 'twentytwelve-block-style', get_template_directory_uri() . '/css/blocks.css', array( 'twentytwelve-style' ), '20230213' );
 
 	// Loads the Internet Explorer specific stylesheet.
```
```suggestion
	// Registers the Internet Explorer specific stylesheet.
```

metodiew replied:
Updated

sabernhardt commented on src/wp-content/themes/twentytwelve/css/ie.css:
```diff
@@ -270,4 +270,4 @@ input[type="password"] {
 	z-index: 1;
-}
\ No newline at end of file
+}
```
Restoring the `ie.css` and `html5.js` added a line break at the end of both files. This is not exactly a problem, but I would prefer not to modify those files at all.

metodiew replied:
it seems like the editor added this by default, restored the original versions as it should be, thank you

sabernhardt reviewed (commented):
I like that you changed "Load" to "Register" in another PR, keeping a verb. Similarly, Twenty Twelve should have "Registers" for its IE stylesheet.

metodiew committed:
Twenty Twelve: update load with register and restore ie.css and html5.js files
